### PR TITLE
Mobile/record sleep scaffold

### PR DESCRIPTION
**Pull request by cicero-mt**
Scaffold for mobile app:

- Dashboard page (home)
- Bluetooth connection page
- Record Sleep guide page

Added automatic routes management with auto_router
High level navigation through an app drawer
Experimenting with chart lib
Renamed and moved some of Mourad's file so conflicts are expected

### CODE REVIEW

File: mobile/lib/src/presentation/navigation/routes/router.gr.dart [hunk 1]
```diff
@@ -0,0 +1,113 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
```
Comment (abelfodil) [inline mobile/lib/src/presentation/navigation/routes/router.gr.dart:1]: Don't track generated files

Add `*.gr.dart` rule to gitignore

Comment (abelfodil) [reply mobile/lib/src/presentation/navigation/routes/router.gr.dart:1]: Add `flutter packages pub run build_runner build --delete-conflicting-outputs` to  `.github/workflows/mobile.yml`

Add `flutter packages pub run build_runner watch`  to readme

File: mobile/pubspec.yaml [hunk 2]
```diff
@@ -46,18 +44,21 @@ dependencies:
 dev_dependencies:
   flutter_test:
     sdk: flutter
+  auto_route_generator: ^0.6.10
+  flutter_launcher_icons: "^0.7.0"
+  build_runner: ^1.10.3
```
Comment (abelfodil) [inline mobile/pubspec.yaml:49]: nit: sort this list

File: mobile/lib/src/application/blocs.dart [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export './device/device_selector_cubit.dart';
+export './eeg_data/data_cubit.dart';
+export './navdrawer/navdrawer_bloc.dart';
```
Comment (abelfodil) [inline mobile/lib/src/application/blocs.dart:3]: We decided not to export packages as a whole. Does this also include blocs? @WilliamHarvey97 @MouradLachhab

File: mobile/lib/src/application/navdrawer/navdrawer_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+part 'navdrawer_event.dart';
+part 'navdrawer_state.dart';
+
+class NavdrawerBloc extends Bloc<NavdrawerEvent, NavdrawerState> {
+  NavdrawerBloc() : super(NavdrawerState.DashBoard);
+
+  @override
+  Stream<NavdrawerState> mapEventToState(
+    NavdrawerEvent event,
+  ) async* {
```
Comment (abelfodil) [inline mobile/lib/src/application/navdrawer/navdrawer_bloc.dart:15]: nit: 
```suggestion
  Stream<NavdrawerState> mapEventToState(NavdrawerEvent event) async* {
```

File: mobile/lib/src/application/navdrawer/navdrawer_event.dart [hunk 5]
```diff
@@ -0,0 +1,19 @@
+part of 'navdrawer_bloc.dart';
+
+abstract class NavdrawerEvent extends Equatable {
+  const NavdrawerEvent();
+  @override
+  List<Object> get props => [];
+}
+
```
Comment (abelfodil) [inline mobile/lib/src/application/navdrawer/navdrawer_event.dart:8]: I would put this in a separate file

**Review by abelfodil — CHANGES_REQUESTED** [review]
A few remarks here:
- The app crashes on start
- Run `flutter format **/*.dart`
- Merge master and fix potential pipeline errors

File: mobile/lib/src/application/blocs.dart (hunk 3 above)
Comment (WilliamHarvey97) [reply mobile/lib/src/application/blocs.dart:3]: In that case, I think it's ok like that. The issue was more about exporting the whole layer.

File: mobile/lib/src/application/eeg_data/data_cubit.dart [hunk 6]
```diff
@@ -1,10 +1,11 @@
 import 'package:polydodo/src/domain/acquisition_device/i_acquisition_device_repository.dart';
 import 'package:polydodo/src/domain/eeg_data/i_eeg_data_repository.dart';
 
-import 'data_states.dart';
 import 'package:bloc/bloc.dart';
```
Comment (WilliamHarvey97) [inline mobile/lib/src/application/eeg_data/data_cubit.dart:4]: packages' import should be placed before project's imports

File: mobile/lib/src/application/navdrawer/navdrawer_bloc.dart [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+
+part 'navdrawer_event.dart';
+part 'navdrawer_state.dart';
+
+class NavdrawerBloc extends Bloc<NavdrawerEvent, NavdrawerState> {
```
Comment (WilliamHarvey97) [inline mobile/lib/src/application/navdrawer/navdrawer_bloc.dart:9]: Pourquoi pas faire un Cubit plutôt qu'un Bloc?

File: mobile/lib/src/presentation/navigation/navdrawer_widget.dart [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+  Widget build(BuildContext context) {
+    return Drawer(
+      child: ListView(
+        padding: EdgeInsets.zero, //only(top: 8.0),
+        children: <Widget>[
+          _createHeader(),
+          _createDrawerItem(
+            icon: Icons.bluetooth,
+            text: "Bluetooth selector",
+            onTap: () {
```
Comment (WilliamHarvey97) [inline mobile/lib/src/presentation/navigation/navdrawer_widget.dart:22]: Create a function like _changeRoute(NavdrawerState state, Routes route) as arguments so code won't need to be changed for each drawerItem.

File: .gitignore [hunk 9]
```diff
@@ -0,0 +1,4 @@
+
+mobile/pubspec.lock
+
+mobile/polydodo-mobile.code-workspace
```
[the remark this answers is not in the document]
Comment (WilliamHarvey97) [reply .gitignore:4]: Also, we don't want to ignore pubspec.lock

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,4 @@
+
+mobile/pubspec.lock
```
[the remark this answers is not in the document]
Comment (WilliamHarvey97) [reply .gitignore:2]: We don't want to ignore pubspec.lock

File: mobile/lib/src/presentation/navigation/routes/router.dart [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+import 'package:polydodo/src/presentation/pages/record_sleep/record_sleep_guide_page.dart';
+
+@MaterialAutoRouter(
+    generateNavigationHelperExtension: true,
+    routes: <AutoRoute>[
+      MaterialRoute(page: NavDrawerPage),
+      MaterialRoute(page: DashBoardPage, initial: true),
+      MaterialRoute(page: RecordSleepGuidePage),
+      MaterialRoute(page: RecordSleepValidatePage),
+      MaterialRoute(page: BluetoothSelectorPage),
```
Comment (WilliamHarvey97) [inline mobile/lib/src/presentation/navigation/routes/router.dart:14]: Why did we renamed Route for Page?

File: .gitignore (hunk 9 above)
[the remark this answers is not in the document]
Comment (abelfodil) [reply .gitignore:4]: https://dart.dev/guides/libraries/private-files

[the remark this answers is not in the document]
Comment (cicero-mt) [reply .gitignore:3]: pour réduire le clutter et avoir juste le dossier /mobile d'ouvert.

File: .gitignore (hunk 10 above)
[the remark this answers is not in the document]
Comment (cicero-mt) [reply .gitignore:1]: conclusion? On a dit qu'on ignore ou pas Pubspec.yaml & .lock? @WilliamHarvey97

File: mobile/lib/src/application/navdrawer/navdrawer_bloc.dart (hunk 7 above)
Comment (cicero-mt) [reply mobile/lib/src/application/navdrawer/navdrawer_bloc.dart:9]: J'étais moins familier avec les cubits. Je peux le changer.

File: mobile/lib/src/presentation/navigation/routes/router.dart (hunk 11 above)
Comment (cicero-mt) [reply mobile/lib/src/presentation/navigation/routes/router.dart:14]: I tought it sounded more smooth to have "routes that point to pages which loads widgets". But I might be completely off.

File: mobile/lib/src/presentation/navigation/navdrawer_widget.dart (hunk 8 above)
Comment (cicero-mt) [reply mobile/lib/src/presentation/navigation/navdrawer_widget.dart]: Not sure how can each drawer item has the same changeRoute call :O ?

File: mobile/lib/src/application/navdrawer/navdrawer_event.dart (hunk 5 above)
Comment (abelfodil) [reply mobile/lib/src/application/navdrawer/navdrawer_event.dart:8]: oh nvm brainfart

File: mobile/.gitignore [hunk 12]
```diff
@@ -40,3 +40,7 @@ app.*.symbols
 
 # Obfuscation related
 app.*.map.json
+
+# Project related
+mobile/polydodo-mobile.code-workspace
```
Comment (abelfodil) [inline mobile/.gitignore:45]:
```suggestion
```

File: mobile/polydodo-mobile.code-workspace [hunk 13]
```diff
@@ -0,0 +1,38 @@
+{
```
Comment (abelfodil) [inline mobile/polydodo-mobile.code-workspace:1]: Remove this file

Comment (abelfodil) [reply mobile/polydodo-mobile.code-workspace:1]: Si on veut ajouter de la config dans `mobile`, faut simplement utiliser le folder `mobile/.vscode`

File: mobile/pubspec.yaml [hunk 14]
```diff
@@ -44,22 +42,25 @@ dependencies:
   streaming_shared_preferences: ^1.0.1
   uuid: ^2.2.2
   usb_serial: ^0.2.4
+  charts_flutter:
```
Comment (abelfodil) [inline mobile/pubspec.yaml:45]: Faut une version précise et aussi sort cette entrée

File: mobile/lib/src/presentation/widgets/simple_line_chart.dart [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+  }
+
+  /// Create one series with sample hard coded data.
+  static List<charts.Series<LinearSales, int>> _createSampleData() {
+    final data = [
+      LinearSales(0, 5),
+      LinearSales(1, 25),
+      LinearSales(2, 100),
+      LinearSales(3, 75),
+    ];
```
Comment (abelfodil) [inline mobile/lib/src/presentation/widgets/simple_line_chart.dart:32]: Put this inside another file for mock data

File: mobile/lib/src/presentation/widgets/simple_line_chart.dart [hunk 16]
```diff
@@ -0,0 +1,52 @@
+/// Example of a simple line chart.
```
Comment (abelfodil) [inline mobile/lib/src/presentation/widgets/simple_line_chart.dart:1]:
```suggestion
```

File: mobile/lib/src/presentation/widgets/simple_line_chart.dart [hunk 17]
```diff
@@ -0,0 +1,52 @@
...
+import 'package:charts_flutter/flutter.dart' as charts;
+import 'package:flutter/material.dart';
+
+class SimpleLineChart extends StatelessWidget {
+  final List<charts.Series> seriesList;
+  final bool animate;
+
+  SimpleLineChart(this.seriesList, {this.animate});
+
+  /// Creates a [LineChart] with sample data and no transition.
```
Comment (abelfodil) [inline mobile/lib/src/presentation/widgets/simple_line_chart.dart:11]:
```suggestion
```

File: mobile/lib/src/presentation/widgets/simple_line_chart.dart [hunk 18]
```diff
@@ -0,0 +1,52 @@
...
+  factory SimpleLineChart.withSampleData() {
+    return SimpleLineChart(
+      _createSampleData(),
+      // Disable animations for image tests.
```
Comment (abelfodil) [inline mobile/lib/src/presentation/widgets/simple_line_chart.dart:15]:
```suggestion
```

File: mobile/lib/src/presentation/widgets/simple_line_chart.dart [hunk 19]
```diff
@@ -0,0 +1,52 @@
...
+        domainFn: (LinearSales sales, _) => sales.year,
+        measureFn: (LinearSales sales, _) => sales.sales,
+        data: data,
+      )
+    ];
+  }
+}
+
+/// Sample linear data type.
+class LinearSales {
```
Comment (abelfodil) [inline mobile/lib/src/presentation/widgets/simple_line_chart.dart:47]:
```suggestion
class SampleLinearSales {
```

Du code autodocumenté est 100x mieux que du code + commentaire, car là tu dois maintenir de l'information à deux places.

File: mobile/lib/src/presentation/navigation/routes/router.dart [hunk 20]
```diff
@@ -0,0 +1,15 @@
+import 'package:auto_route/auto_route_annotations.dart';
+import 'package:polydodo/src/presentation/navigation/navdrawer_widget.dart';
```
Comment (abelfodil) [inline mobile/lib/src/presentation/navigation/routes/router.dart:2]:
```suggestion
```

**Review by abelfodil — CHANGES_REQUESTED** [review]

File: mobile/pubspec.yaml (hunk 14 above)
Comment (cicero-mt) [reply mobile/pubspec.yaml]: Inséré dans la pull request sans le vouloir. Ça sera pour une autre PR.

File: mobile/lib/src/presentation/widgets/simple_line_chart.dart (hunk 19 above)
Comment (cicero-mt) [reply mobile/lib/src/presentation/widgets/simple_line_chart.dart]: C'est le sample code du package, ça devait pas être dans la PR. Mais 👍🏼

File: mobile/lib/src/presentation/navigation/navdrawer_widget.dart (hunk 8 above)
Comment (WilliamHarvey97) [reply mobile/lib/src/presentation/navigation/navdrawer_widget.dart]: `onTap: _changeRoute(NavdrawerState.BluetoothSelector, Routes.bluetoothSelectorPage),`

Comment (WilliamHarvey97) [reply mobile/lib/src/presentation/navigation/navdrawer_widget.dart]: And you'll need to define this:
```
void _changeRoute(BuildContext context, NavdrawerState nextState, Routes nextRoute) {
          context
              .bloc<NavdrawerBloc>()
              .add(NavdrawerUpdated(NavdrawerState.BluetoothSelector));
          ExtendedNavigator.of(context).replace(Routes.bluetoothSelectorPage);
}
```

Comment (cicero-mt) [reply mobile/lib/src/presentation/navigation/navdrawer_widget.dart]: Hum you seem to be refering to an outdated version of the file

**Review by abelfodil — APPROVED** [review]

**Pull request merged**